packages/cozy-harvest-lib/src/datacards/FileDataCard.jsx

```diff
@@ -165,17 +165,14 @@ const FileCard = ({ files, loading, konnector, trigger }) => {
   )
 }
 
-const makeQueryFromProps = ({ accountId }) => ({
+const makeQueryFromProps = ({ accountId, trigger }) => ({
   query: Q('io.cozy.files')
     .where({
-      'cozyMetadata.sourceAccount': accountId,
+      dir_id: trigger.message.folder_to_save,
       trashed: false
     })
-    .indexFields(['cozyMetadata.sourceAccount', 'cozyMetadata.createdAt'])
-    .sortBy([
-      { 'cozyMetadata.sourceAccount': 'desc' },
-      { 'cozyMetadata.createdAt': 'desc' }
-    ])
+    .indexFields(['dir_id', 'cozyMetadata.createdAt'])
```

Comment:
I think you can also index by trashed

Reply:
I am not sure ? Is it really necessary, I do not think that a filter on the documents already filtered by the other conditions is going to take long ? What do you think @paultranvan ? Should we put all the fields always ?

Reply:
For me it's ok not indexing it, as long as the percent of trashed docs remains low. In a worst case scenario, you might have all your docs trashed, and CouchDB will have to full scan the index to finally return nothing.
A possible alternative here is a partial index : 
```javascript
  query: Q('io.cozy.files')
    .where({
      dir_id: trigger.message.folder_to_save
    })
    .partialIndex(
        $or: [
          {
            trashed: {
              $exists: false
            }
          },
          {
            trashed: false
          }
        ]
      })
    .indexFields(['dir_id', 'cozyMetadata.createdAt'])
    .sortBy([{ dir_id: 'desc' }, { 'cozyMetadata.createdAt': 'desc' }])

 ```
 
 This way, the docs with a `trashed: true` field won't be indexed at all

Reply:
BTW, we quite often need to filter out trashed docs, and this partial filter is a bit verbose, maybe we could add some helper on the dsl level like `.filterTrashed()` to handle this?

Reply:
OK thanks for the explanation, I think the state of the PR is ready then.

@paultranvan I've put your suggestion in an issue along with an idea : https://github.com/cozy/cozy-client/issues/909